sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/postgres/PostgresInt4Tests.java

```diff
@@ -0,0 +1,419 @@
+                        " -2147483647\n" +
+                        "(1 row)"
+        );
+    }
+
+    // This passes for the Calcite version but fails for the run time version
+    // Check PostgresInt2Tests::testINT2MINOverflow for details
+    @Test @Ignore
+    public void testINT4MINOverflow() {
+        this.q(
+                "SELECT (-2147483648)::int4 % (-1)::int4 as x;\n" +
+                        " x \n" +
+                        "---\n" +
+                        "  0"
+        );
+    }
+
+    @Test @Ignore
+    public void testINT4MINOverflowError() {
+        String error = "INT4 out of range";
+
+        this.shouldFail("SELECT (-2147483648)::int4 * (-1)::int2", error);
+        this.shouldFail("SELECT (-2147483648)::int4 / (-1)::int2", error);
+    }
+
+    @Test @Ignore("fails because we round differently than Postgres")
+    public void testFloatRound() {
+        this.q(
+                "SELECT x, x::int4 AS int4_value\n" +
+                        "FROM (VALUES (-2.5::float8),\n" +
+                        "             (-1.5::float8),\n" +
+                        "             (-0.5::float8),\n" +
+                        "             (0.0::float8),\n" +
+                        "             (0.5::float8),\n" +
+                        "             (1.5::float8),\n" +
+                        "             (2.5::float8)) t(x);\n" +
+                        "  x   | int4_value \n" +
+                        "------+------------\n" +
+                        " -2.5 |         -2\n" +
+                        " -1.5 |         -2\n" +
+                        " -0.5 |          0\n" +
+                        "    0 |          0\n" +
+                        "  0.5 |          0\n" +
+                        "  1.5 |          2\n" +
+                        "  2.5 |          2"
+        );
+    }
+    @Test @Ignore("fails because we round differently than Postgres")
+    public void testNumericRound() {
+        this.q(
+                "SELECT x, x::int4 AS int4_value\n" +
+                        "FROM (VALUES (-2.5::numeric),\n" +
+                        "             (-1.5::numeric),\n" +
+                        "             (-0.5::numeric),\n" +
+                        "             (0.0::numeric),\n" +
+                        "             (0.5::numeric),\n" +
+                        "             (1.5::numeric),\n" +
+                        "             (2.5::numeric)) t(x);\n" +
+                        "  x   | int4_value \n" +
+                        "------+------------\n" +
+                        " -2.5 |         -3\n" +
+                        " -1.5 |         -2\n" +
+                        " -0.5 |         -1\n" +
+                        "  0.0 |          0\n" +
+                        "  0.5 |          1\n" +
+                        "  1.5 |          2\n" +
+                        "  2.5 |          3"
+        );
+    }
+
+    @Test @Ignore("GCD not supported yet")
```

Comment:
neither does calcite.
We can add it there, or we can do it as a user-defined function, or we can just skip it.
For now I would skip it.